lib/rules/brace-style.js

```diff
@@ -67,21 +67,31 @@ module.exports = function(context) {
                 previousToken = sourceCode.getTokenBefore(block);
                 curlyToken = sourceCode.getFirstToken(block);
                 curlyTokenEnd = sourceCode.getLastToken(block);
-                curlyTokensOnSameLine = curlyToken.loc.start.line === curlyTokenEnd.loc.start.line;
+                allOnSameLine = previousToken.loc.start.line === curlyToken.loc.start.line &&
+                    curlyToken.loc.start.line === curlyTokenEnd.loc.start.line;
+
+                if (allOnSameLine && params.allowSingleLine) {
+                    return;
+                }
 
                 if (style !== "allman" && previousToken.loc.start.line !== curlyToken.loc.start.line) {
                     context.report(node, OPEN_MESSAGE);
-                } else if (style === "allman" && previousToken.loc.start.line === curlyToken.loc.start.line && !params.allowSingleLine) {
+                } else if (
+                    style === "allman" &&
+                    previousToken.loc.start.line === curlyToken.loc.start.line
+                ) {
                     context.report(node, OPEN_MESSAGE_ALLMAN);
                 }
 
-                if (!block.body.length || curlyTokensOnSameLine && params.allowSingleLine) {
+                if (!block.body.length) {
                     return;
                 }
 
                 if (curlyToken.loc.start.line === block.body[0].loc.start.line) {
                     context.report(block.body[0], BODY_MESSAGE);
-                } else if (curlyTokenEnd.loc.start.line === block.body[block.body.length - 1].loc.start.line) {
+                }
+
+                if (curlyTokenEnd.loc.start.line === block.body[block.body.length - 1].loc.start.line) {
```

Comment:
@nzakas - I did modify tests (added the masked errors) for them to pass with this change.